domain/permission/state/state.go

```diff
@@ -30,14 +29,17 @@ type Logger interface {
 // State describes retrieval and persistence methods for storage.
 type State struct {
 	*domain.StateBase
-	logger Logger
+	sharedState *SharedState
```

Comment:
Could this be embedded to allow users of permissions state to access the methods without the indirection? (ignore this if the shared methods are meant to be private)

Reply:
fwiw, I *usually* regret embedding. It tends to lead to far more confusion about what is going on.

Reply:
I'm happy to move.